src/Microsoft.Health.Fhir.Api/Features/Routing/SearchPostReroutingMiddleware.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Health.Fhir.Api.Features.Routing
+{
+    public class SearchPostReroutingMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public SearchPostReroutingMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var request = context.Request;
+
+            if (request != null
+                && request.Method == "POST"
+                && request.Path.Value.EndsWith(KnownRoutes.Search, System.StringComparison.OrdinalIgnoreCase)
+                && request.ContentType == "application/x-www-form-urlencoded")
```

Comment:
If a user makes a request with the wrong/missing content type they get a 404 back. I feel this should be a 400 informing them of the wrong content type.